### ISSUE TITLE
Some concerns about the test.

### ISSUE THREAD

**Issue by joshyZhou** [linYDTHU/DesnowNet_Context-Aware_Deep_Network_for_Snow_Removal#2]
Hi,
Thanks for sharing the code of your great work! Here, we wonder about the detailed setting of testing on Snow 100K dataset. (1)Is this test process implemented on the original resolution or 64 \times 64 or 256 \times 256, or something else? (2) is the calculation of PSNR and SSIM performed on RGB or Y channel?
Looking forward to your answer!
Best,
josh

**Comment by linYDTHU** [comment]
Hi, thanks for your interest in this repo. 1)I just checked the code and it seems that the test process is implemented on 64 \times 64. You can change it to the original resolution by setting the `snow_dataset(gt_root, mask_root, synthetic_root, is_crop=False)`. 2)As presented in the paper, PSNR and SSIM should be calculated between the snow-free output \hat{y} and the corresponding
ground truth y.  Therefore, it's performed on RGB channels. The PSNR of z_hat is just to evaluate the quality of predicted snow masks. 
Btw, this is a project I did one year ago, so there's a lot of implementation details I cannot remember correctly. If u have further questions or find bugs, feel free to submit a new issue or a pull request.

**Comment by joshyZhou** [comment]
> Hi, thanks for your interest in this repo. 1)I just checked the code and it seems that the test process is implemented on 64 \times 64. You can change it to the original resolution by setting the `snow_dataset(gt_root, mask_root, synthetic_root, is_crop=False)`. 2)As presented in the paper, PSNR and SSIM should be calculated between the snow-free output \hat{y} and the corresponding ground truth y. Therefore, it's performed on RGB channels. The PSNR of z_hat is just to evaluate the quality of predicted snow masks. Btw, this is a project I did one year ago, so there's a lot of implementation details I cannot remember correctly. If u have further questions or find bugs, feel free to submit a new issue or a pull request.

Thanks for your quick reply! It is clear to me now. 
Besides, we wonder (1) if the test size of 64 x 64 is consistent with the original work since we have no idea of the specific settings.  (2) when we want to reproduce the numbers reported in the published paper, how can we to set the hyperparameters correctly, for instance, the iterations?
Best,
josh

**Comment by linYDTHU** [comment]
1) The paper indicates that the ablation studies are conducted on the original resolution. 2)Unluckily, I just followed the configurations reported in the paper and could not fully reproduce the results last year. I checked the code twice and was unable to find an obvious bug that time. You can send an email to the authors of this paper to request for the code or advice.  I am also willing to accept any useful pull request that can make this repo better. Btw, I did some improvements based on these code, including two-stage training or predicting the product of z and a. Hope they can help u to explore more in this topic. I will revisit this repo when I have time. Best wishes!

**Comment by joshyZhou** [comment]
> 1. The paper indicates that the ablation studies are conducted on the original resolution. 2)Unluckily, I just followed the configurations reported in the paper and could not fully reproduce the results last year. I checked the code twice and was unable to find an obvious bug that time. You can send an email to the authors of this paper to request for the code or advice.  I am also willing to accept any useful pull request that can make this repo better. Btw, I did some improvements based on these code, including two-stage training or predicting the product of z and a. Hope they can help u to explore more in this topic. I will revisit this repo when I have time. Best wishes!

Thanks for your kind reply and advice. I will give it a retry!
Best,
josh
